test/e2e/gameserverallocation_test.go

```diff
@@ -720,6 +845,110 @@ func TestCounterGameServerAllocationActions(t *testing.T) {
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		t.SkipNow()
+	}
+	t.Parallel()
+	ctx := context.Background()
+	client := framework.AgonesClient.AgonesV1()
+
+	lists := map[string]agonesv1.ListStatus{}
+	lists["players"] = agonesv1.ListStatus{
+		Values:   []string{"player0", "player1", "player2"},
+		Capacity: 8,
+	}
+
+	flt := defaultFleet(framework.Namespace)
+	flt.Spec.Template.Spec.Lists = lists
+
+	flt, err := client.Fleets(framework.Namespace).Create(ctx, flt.DeepCopy(), metav1.CreateOptions{})
+	require.NoError(t, err)
+	defer client.Fleets(framework.Namespace).Delete(ctx, flt.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint:errcheck
+	framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+
+	fleetSelector := metav1.LabelSelector{MatchLabels: map[string]string{agonesv1.FleetNameLabel: flt.ObjectMeta.Name}}
+	allocated := agonesv1.GameServerStateAllocated
+	ready := agonesv1.GameServerStateReady
+	one := int64(1)
+
+	testCases := map[string]struct {
+		gsa          allocationv1.GameServerAllocation
+		wantGsaErr   bool
+		wantCapacity *int64
+		wantValues   []string
+	}{
+		"add List values": {
+			gsa: allocationv1.GameServerAllocation{
+				Spec: allocationv1.GameServerAllocationSpec{
+					Selectors: []allocationv1.GameServerSelector{
+						{LabelSelector: fleetSelector},
+					},
+					Lists: map[string]allocationv1.ListAction{
+						"players": {
+							AddValues: []string{"player3", "player4", "player5"},
+						}}}},
+			wantGsaErr: false,
+			wantValues: []string{"player0", "player1", "player2", "player3", "player4", "player5"},
+		},
+		"change List capacity truncates": {
+			gsa: allocationv1.GameServerAllocation{
+				Spec: allocationv1.GameServerAllocationSpec{
+					Selectors: []allocationv1.GameServerSelector{
+						{LabelSelector: fleetSelector},
+					},
+					Lists: map[string]allocationv1.ListAction{
+						"players": {
+							Capacity: &one,
+						}}}},
+			wantGsaErr: false,
+			wantValues: []string{"player0"},
+		},
+	}
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+
+			gsa, err := framework.AgonesClient.AllocationV1().GameServerAllocations(flt.ObjectMeta.Namespace).Create(ctx, testCase.gsa.DeepCopy(), metav1.CreateOptions{})
+			if testCase.wantGsaErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			assert.Equal(t, string(allocated), string(gsa.Status.State))
+
+			gs1, err := framework.AgonesClient.AgonesV1().GameServers(flt.ObjectMeta.Namespace).Get(ctx, gsa.Status.GameServerName, metav1.GetOptions{})
+			require.NoError(t, err)
+			assert.Equal(t, allocated, gs1.Status.State)
+			assert.NotNil(t, gs1.ObjectMeta.Annotations["agones.dev/last-allocated"])
+
+			list, ok := gs1.Status.Lists["players"]
+			assert.True(t, ok)
+			if testCase.wantCapacity != nil {
+				assert.Equal(t, *testCase.wantCapacity, list.Capacity)
+			}
+			if diff := cmp.Diff(testCase.wantValues, list.Values); diff != "" {
```

Comment:
Any reason not `assert.Equals` or `require.Equals` ? Then you get a nice diff between the two values as well if they aren't Equal.

Reply:
I was thinking that assert.Equal couldn't handle slices, but apparently it works with string slices. Changing to assert.Equal.